### DISCUSSION TITLE
Feature Request: GitHub Actions Workflow CLI linter - catch basic security errors too

### DISCUSSION THREAD

**Discussion by HariSekhon** [community/community#18131] [Actions]
Feature Request for a CLI to lint GitHub Actions Workflow files.

Right now I lint the yaml for correctness, but I'd like to lint for more sophisticated aspects of the GitHub Actions workflow, such as invalid constructs/keywords out of place and especially checking that inputs, env, secrets, and other contexts are not passed straight into run shell steps which risk script injection, to make sure they are always passed through an environment variable instead to be properly escaped.

**Comment by roryabraham** [comment]
There are a few things we do to aid with this:

1. https://github.com/rhysd/actionlint is an awesome package that works quite well.
1. We also validate the action and workflow schemas using [SchemaStore](https://www.schemastore.org/json/).

You can see our validation script [here](https://github.com/Expensify/App/blob/aec6bc2a328accc36dfed9140b8a2e9e60d9b70b/.github/scripts/validateActionsAndWorkflows.sh) and it works locally, maybe some of that would be useful to you?

Next step for us would be to set up a yaml style linter to promote consistency in our yaml style
